src/main/java/org/fundacionjala/virtualassistant/player/spotify/client/SpotifyClient.java

```diff
@@ -0,0 +1,364 @@
+            JsonNode jsonNode = objectMapper.readTree(playerData);
+
+            if (jsonNode.has("progress_ms")) {
+                return jsonNode.get("progress_ms").asInt();
+            }
+
+            return 0;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return 0;
+        }
+    }
+
+    public void pauseSongOnDevice(String trackUri) {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        String requestBody = "{\"uris\": [\"" + trackUri + "\"]}";
+        HttpEntity<String> entity = new HttpEntity<>(requestBody, headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/player/pause",
+                HttpMethod.PUT,
+                entity,
+                String.class
+        );
+    }
+
+    public boolean playNextTrackOnDevice() {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+
+        HttpEntity<String> entity = new HttpEntity<>(headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/player/next",
+                HttpMethod.POST,
+                entity,
+                String.class
+        );
+
+        return response.getStatusCode() == HttpStatus.NO_CONTENT;
+    }
+
+    public boolean playPreviousTrackOnDevice() {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+
+        HttpEntity<String> entity = new HttpEntity<>(headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/player/previous",
+                HttpMethod.POST,
+                entity,
+                String.class
+        );
+
+        return response.getStatusCode() == HttpStatus.NO_CONTENT;
+    }
+
+    @Override
+    public void logout() {
+        accessToken = null;
+    }
+
+    public String searchTrackByArtistAndTrack(String artist, String track) {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+
+        HttpEntity<String> entity = new HttpEntity<>(headers);
+
+        String searchQuery = "https://api.spotify.com/v1/search?q=" + artist + " " + track + "&type=track&limit=1";
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                searchQuery,
+                HttpMethod.GET,
+                entity,
+                String.class
+        );
+
+        if (response.getStatusCode() == HttpStatus.OK) {
```

Comment:
this will thrown a null pointer exceptio if the request is null, please verify if the request is null

Reply:
Warnings appear because response is always not null